### PR TITLE
Doc: update the building guide for Linux

### PR DESCRIPTION
**Pull request by Zhangbaowen-Hashdata**
closes: #ISSUE_Number

---

### Change logs

Since the previous document is a legacy document, much of the code may be invalid, so we verified and tidied it up. The accuracy of the document is ensured, which is easy for users to read, and the environment can be quickly built according to the document.

### Why are the changes needed?

Ensure that the documentation is accurate and usable

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

According to the documents needed by the users can quickly build on Linux environment

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to @cloudberrydb/dev team for review and approval when your PR is ready🥳

### CODE REVIEW

**Comment by Zhangbaowen-Hashdata** [comment]
PR has been generated, please help me have a look at what's the problem, thank you@tuhaihe @tuhaihe

File: readmes/README.Linux.md [hunk 1]
```diff
@@ -1,108 +1,173 @@
-## For CentOS:
+# Set up development environment for Cloudberry Database on Linux
 
-- Install Dependencies
+This document shares how to set up the development environments on different Linux systems (CentOS 7, RHEL, and Ubuntu). Before you start to [explore the Cloudberry Database](https://hackmd.io/XIgu6X8TTLWFmCttr2LQ6Q?view), follow this document to download and install the required tools and dependencies, and create demo user.
```
Comment (tuhaihe) [inline readmes/README.Linux.md:3]: > [explore the Cloudberry Database](https://hackmd.io/XIgu6X8TTLWFmCttr2LQ6Q?view)
The link is wrong, please update it.

**Comment by tuhaihe** [comment]
Hi @Zhangbaowen-Hashdata: Thanks for your contribution. Please squash your commits down to one before this PR is merged. Maybe you can take our [coding guide](https://cloudberrydb.org/contribute/code#contribution-workflow) as a reference.

File: readmes/README.Linux.md (hunk 1 above)
Comment (Zhangbaowen-Hashdata) [reply readmes/README.Linux.md:3]: Thank you for your correction. May I ask what changes should be made to this link?

Comment (tuhaihe) [reply readmes/README.Linux.md:3]: Hi @TomShawn, any ideas?

Comment (Zhangbaowen-Hashdata) [reply readmes/README.Linux.md]: With @TomShawn assistance, I have corrected the link.

File: readmes/README.Linux.md [hunk 2]
```diff
@@ -1,108 +1,173 @@
...
-  sudo yum install -y centos-release-scl
-  sudo yum install -y devtoolset-7-toolchain
-  echo 'source scl_source enable devtoolset-7' >> ~/.bashrc
-  ```
+Install dependencies according to your operating systems:
 
-## For RHEL:
+- [For CentOS 7](#For-CentOS-7)
+- [For RHEL 7 and 8](#For-RHEL-7-and-8)
+- [For Ubuntu 18.04 or later](#For-Ubuntu-1804-or-later)
```
Comment (TomShawn) [inline readmes/README.Linux.md:21]:
```suggestion
- [For CentOS 7](#for-centos-7)
- [For RHEL 7 and 8](#for-rhel-7-and-8)
- [For Ubuntu 18.04 or later](#for-ubuntu-1804-or-later)
```

Comment (Zhangbaowen-Hashdata) [reply readmes/README.Linux.md:21]: Okay, I will convert the letters to lowercase as required

**Review by tuhaihe — APPROVED** [review]
@tuhaihe LGTM.

**Review by TomShawn — APPROVED** [review]
LGTM

**Comment by tuhaihe** [comment]
Hi @baotingfang @my-ship-it, please help review this PR, thanks.

**Review by baotingfang — APPROVED** [review]
LGTM

**Review by my-ship-it — APPROVED** [review]

**Comment by my-ship-it** [comment]
LGTM

**Pull request merged**